### ISSUE TITLE
Missing volume in workflow spec when generated from workflow template fired from sensor trigger

### ISSUE THREAD

**Issue by jmeridth** [argoproj/argo-events#1542]
**Describe the bug**

### TL;DR

Sensor -> WorkflowTemplate -> Workflow

Error: `task 'webhook-f4hmg[0].template1-step.step1' errored: volume 'blah-blah-blah' not found in workflow spec`

Workflow is missing volume section in spec that is present in the WorkflowTemplate

This does not happen when workflow is created from workflowtemplate via UI or CLI

### LONGER VERSION

Sensor that uses a templateRef (workflow template) to create a workflow would exclude the volume section at the top of the workflow template from the generated workflow?  This only happens when we do sensor -> workflowtemplate -> workflow.  If we create the workflow from the workflowtemplate from the UI or CLI the volume section is there and can be used.

versions:
argo-events v1.5.6
argo-workflows v3.2.6

error we get when going through argo-events is: 

`task 'webhook-f4hmg[0].template1-step.step1' errored: volume 'blah-blah-blah' not found in workflow spec`

triggers section of our sensor.yaml

```
  triggers:
    - template:
        name: webhook-workflow-trigger
        k8s:
          operation: create
          source:
            resource:
              apiVersion: argoproj.io/v1alpha1
              kind: Workflow
              metadata:
                generateName: webhook-
              spec:
                serviceAccountName: workflow-pods-sa
                entrypoint: template1-run
                arguments:
                  parameters:
                    - name: sha
                    - name: ref
                templates:
                  - name: template1-run
                    steps:
                      - - name: template1-step
                          templateRef:
                            name: template1
                            template: main
```

spec section of our workflowtemplate.yaml

```
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: template1
spec:
  arguments:
    parameters:
    - name: sha
      value: "abcdefgh"
    - name: ref
      value: "master"
  serviceAccountName: workflow-sa
  entrypoint: main
  imagePullSecrets:
    - name: image-pull-secret
  volumes:
  - name: ssh-git-creds
    secret:
      defaultMode: 0600
      secretName: ssh-git-creds
      items:
        - key: ssh_privatekey
          path: id_rsa
        - key: known_hosts
          path: known_hosts
  templates:
```

when we curl the argo-events endpoint (no issues) and the sensor triggers, the resulting workflow does not have the volumes section in its spec.  Again, using same workflowtemplate through UI or CLI works fine, volume section in spec.
Thank you for any direction you can give.  We did see #342 but an upgrade was the suggested solution.  we're well past that suggested version.

**To Reproduce**
1. use curl to call argo-events endpoint
2. see error in Argo Workflow Dashboard UI `task 'webhook-f4hmg[0].template1-step.step1' errored: volume 'blah-blah-blah' not found in workflow spec`

**Expected behavior**
Successful run of the workflow because volume is found in spec of generated workflow from workflowTemplate

**Screenshots**

**Environment (please complete the following information):**
 - Kubernetes: v1.22
 - Argo Workflows: v3.2.6
 - Argo Events: v1.5.6

**Additional context**
[This was mentioned in the CNCF #argo-events slack yesterday](https://cloud-native.slack.com/archives/C01TNKD6KL6/p1642634709037400)

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by whynowy** [comment]
What `k8s` trigger with `create` operation does is exactly the same as a `kubectl create` command with the workflow spec. Could you try with the command line to see if it works? @jmeridth
